packages/opentelemetry-api/src/api/context.ts

```diff
@@ -45,15 +52,25 @@ export class ContextAPI {
   public setGlobalContextManager(
     contextManager: ContextManager
   ): ContextManager {
-    this._contextManager = contextManager;
+    if (_global[GLOBAL_CONTEXT_MANAGER_API_KEY]) {
```

Comment:
Yes it can only be set once. There is also a `disable` method to unset it so it can be set again. This is to guard against the case where multiple APIs attempt to register themselves, resulting in a version mismatch.

Reply:
why is then then returning the Noop instead of the one that has been already set ?

Reply:
I can make it return `this._getContextManager()`